src/components/Workshop/Workshop.test.js

```diff
@@ -1,20 +1,149 @@
 import React from 'react'
-import { shallow } from 'enzyme'
+import { screen, render } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import FarmhandContext from '../Farmhand/Farmhand.context'
+
+import { getUpgradesAvailable } from './getUpgradesAvailable'
 
 import Workshop from './Workshop'
 
-let component
+jest.mock('./getUpgradesAvailable')
 
-beforeEach(() => {
-  component = shallow(
-    <Workshop
-      {...{
-        learnedRecipes: {},
-      }}
-    />
-  )
-})
+jest.mock('../../data/maps', () => ({
+  ...jest.requireActual('../../data/maps'),
+  recipesMap: {
+    'kitchen-recipe-1': {
+      id: 'kitchen-recipe-1',
+      name: 'kitchen recipe 1',
+      recipeType: 'KITCHEN',
+    },
+    'kitchen-recipe-2': {
+      id: 'kitchen-recipe-2',
+      name: 'kitchen recipe 2',
+      recipeType: 'KITCHEN',
+    },
+    'kitchen-recipe-3': {
+      id: 'kitchen-recipe-3',
+      name: 'kitchen recipe 3',
+      recipeType: 'KITCHEN',
+    },
+    'forge-recipe-1': {
+      id: 'forge-recipe-1',
+      name: 'forge recipe 1',
+      recipeType: 'FORGE',
+    },
+    'forge-recipe-2': {
+      id: 'forge-recipe-2',
+      name: 'forge recipe 2',
+      recipeType: 'FORGE',
+    },
+  },
+}))
+
+jest.mock('../Recipe', () => () => <div data-testid="Recipe">Recipe</div>)
+jest.mock('../UpgradePurchase', () => () => (
+  <div data-testid="UpgradePurchase">UpgradePurchase</div>
+))
+
+describe('<Workshop />', () => {
+  let gameState = {
+    learnedRecipes: {},
+    purchasedSmelter: false,
+    toolLevels: {},
+  }
+
+  const renderWorkshop = gameState => {
+    render(
+      <FarmhandContext.Provider value={{ gameState, handlers: {} }}>
+        <Workshop />
+      </FarmhandContext.Provider>
+    )
+  }
+
+  describe('kitchen', () => {
+    let numLearnedRecipes = 0
+    beforeEach(() => {
+      numLearnedRecipes = Object.keys(gameState.learnedRecipes).length
```

Comment:
Should this line go below the `gameState.learnedRecipes` on line 69? It seems to be using stale data here.

Reply:
good catch!